lib/streamlit/elements/image.py

```diff
@@ -288,22 +287,22 @@ def _ensure_image_size_and_format(
     MAXIMUM_CONTENT_WIDTH. Ensure the image's format corresponds to the given
     ImageFormat. Return the (possibly resized and reformatted) image bytes.
     """
-    image = Image.open(io.BytesIO(image_data))
-    actual_width, actual_height = image.size
+    pil_image = Image.open(io.BytesIO(image_data))
+    actual_width, actual_height = pil_image.size
 
     if width < 0 and actual_width > MAXIMUM_CONTENT_WIDTH:
         width = MAXIMUM_CONTENT_WIDTH
 
     if width > 0 and actual_width > width:
         # We need to resize the image.
         new_height = int(1.0 * actual_height * width / actual_width)
-        image = image.resize((width, new_height), resample=Image.BILINEAR)
-        return _PIL_to_bytes(image, format=image_format, quality=90)
+        pil_image = pil_image.resize((width, new_height), resample=Image.BILINEAR)
+        return _PIL_to_bytes(pil_image, format=image_format, quality=90)
 
-    ext = imghdr.what(None, image_data)
-    if ext != image_format.lower():
+    # pil_image.format should be defined since we call Image.open()
+    if pil_image.format != image_format:
```

Comment:
nit: don't we still want to call `image_format.lower()`?

Reply:
so when I tested it in pr description, `imghdr` returned a lowercase version but pillow actually returns an uppercase version.